test-data/unit/cmdline.test

```diff
@@ -681,3 +681,178 @@ from unreal import F
 
 def f(x: F) -> None: pass
 [out]
+
+[case testDisallowAnyExplicitDefSignature]
+# cmd: mypy m.py
+[file mypy.ini]
+[[mypy]
+[[mypy-m*]
+disallow_any = explicit
+
+[file m.py]
+from typing import Any, List
+
+def f(x: Any) -> None:
+    pass
+
+def g() -> Any:
+    pass
+
+def h() -> List[Any]:
+    pass
+
+[out]
+m.py:3: error: Explicit "Any" is not allowed in a type annotation
+m.py:6: error: Explicit "Any" is not allowed in a type annotation
+m.py:9: error: Explicit "Any" is not allowed in a type annotation
+
+[case testDisallowAnyExplicitVarDeclaration]
+# cmd: mypy m.py
+
+[file mypy.ini]
+[[mypy]
+[[mypy-m*]
+disallow_any = explicit
+
+[file m.py]
+from typing import Any, List
+v: Any = ''
+w = ['']  # type: List[Any]
```

Comment:
All right, that makes a lot of sense.

Also, I will split these tests into two.